.github/workflows/deploy_ios_app.yaml

```diff
@@ -0,0 +1,106 @@
+      flutter_version:
+        type: string
+        default: ''
+        required: false
+        description: 'The flutter version used to build the project'
+        
+    secrets:
+      appstore_issuer_id:
+        required: true
+      appstore_cert_base64:
+        required: true
+      appstore_cert_password:
+        required: true
+      mobile_provision_base64:
+        required: true
+      keychain_password:
+        required: true
+      appstore_api_key_id:
+        required: true
+      appstore_api_key_base64:
+        required: true
+       
+jobs:
+  deploy_ios_app:
+    defaults:
+      run:
+        working-directory: ${{ inputs.project_path }}
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install the Apple certificate and provisioning profile
+        env:
+          BUILD_CERTIFICATE_BASE64: ${{ secrets.appstore_cert_base64 }}
+          P12_PASSWORD: ${{ secrets.appstore_cert_password }}
+          BUILD_PROVISION_PROFILE_BASE64: ${{ secrets.mobile_provision_base64 }}
+          KEYCHAIN_PASSWORD: ${{ secrets.keychain_password }}
+        run: |
+          # Create variables
+          CERTIFICATE_PATH=$RUNNER_TEMP/build_certificate.p12
+          PP_PATH=$RUNNER_TEMP/build_pp.mobileprovision
+          KEYCHAIN_PATH=$RUNNER_TEMP/app-signing.keychain-db
+          
+          # Import certificate and provisioning profile from secrets
+          echo -n "$BUILD_CERTIFICATE_BASE64" | base64 --decode --output $CERTIFICATE_PATH
+          echo -n "$BUILD_PROVISION_PROFILE_BASE64" | base64 --decode --output $PP_PATH
+          
+          # Create temporary keychain
+          security create-keychain -p "$KEYCHAIN_PASSWORD" $KEYCHAIN_PATH
+          security set-keychain-settings -lut 21600 $KEYCHAIN_PATH
+          security unlock-keychain -p "$KEYCHAIN_PASSWORD" $KEYCHAIN_PATH
+          
+          # Import certificate to keychain
+          security import $CERTIFICATE_PATH -P "$P12_PASSWORD" -A -t cert -f pkcs12 -k $KEYCHAIN_PATH
+          security list-keychain -d user -s $KEYCHAIN_PATH
+          
+          # Apply provisioning profile
+          mkdir -p ~/Library/MobileDevice/Provisioning\ Profiles
+          cp $PP_PATH ~/Library/MobileDevice/Provisioning\ Profiles
+      - name: Install App Store Api Key
+        run: |
+          mkdir ~/private_keys
+          echo -n "${{ secrets.appstore_api_key_base64 }}" | base64 --decode --output ~/private_keys/AuthKey_${{ secrets.appstore_api_key_id }}.p8
+      - uses: subosito/flutter-action@v2
+        with:
+          flutter-version: ${{ inputs.flutter_version }}
+          channel: "stable"
+          cache: true
+          cache-key: flutter-:os:-:channel:-:version:-:arch:-:hash:-${{ hashFiles('**/pubspec.lock') }}
+      - run: dart --version
+      - run: flutter --version
+      - name: Install Dependencies
+        run: |
+          flutter clean
+          flutter pub get
+      - name: Build iOS app
```

Comment:
@DDavidPrime @PrimeSwiftArt have you considered using fastlane as a base for the GitHub as well as Azure DevOps distribution? 

From what I see the build and deploy phases are already built in Fastlane and we can use them out of the box instead of composing low level Xcode commands. 

Additionally Fastlane offers actions for provisioning profile as well as the needed certificate.

Reply:
The other option is to look at the  already built action on GitHub - https://github.com/marketplace/actions/apple-provisioning-profile-action